cassiopeia/simulator/TreeSimulator.py

```diff
@@ -0,0 +1,19 @@
+"""
+Abstract class TreeSimulator, for tree simulation module.
+
+All tree simulators are derived classes of this abstract class, and at a minimum
+implement a method called `simulate_tree`.
+"""
+import abc
+from cassiopeia.data import CassiopeiaTree
+
+
+class TreeSimulator(abc.ABC):
+    """
+    TreeSimulator is an abstract class that all tree simulators derive from.
+    """
+
+    @abc.abstractmethod
+    def simulate_tree(self) -> CassiopeiaTree:
+        """Simulate a CassiopeiaTree"""
```

Comment:
Could you please add a bit more documentation about the `simulate_tree` method? What considerations should a user have about this function, and what do they expect to get out of the function call? What fields will be populated in the `CassiopeiaTree`, and which ones would not be?

Reply:
Added a minimal description, although I don't know how much to say here in `simulate_tree` vs in the class' docstring.